includes/subscriptions/class-wc-payments-invoice-service.php

```diff
@@ -169,57 +169,25 @@ public function delete_invoice_items( array $invoice_item_ids ) {
 	 *
 	 * @param WC_Subscription $subscription The subscription.
 	 *
-	 * @return array|WP_Error Invoice item data or WP_Error.
+	 * @return array Invoice item data.
 	 */
 	private function prepare_invoice_item_data( WC_Subscription $subscription ) {
-		$data       = [];
-		$items      = [];
-		$is_delayed = false;
-		$is_new     = $subscription->get_parent()->needs_payment();
-
-		// We need to get the discount from the parent order to check for signup fee coupon when subscription is new.
-		$discount = $is_new ? $subscription->get_parent()->get_total_discount( false ) : $subscription->get_total_discount( false );
+		$data     = [];
+		$discount = $subscription->get_total_discount( false );
+		$currency = $subscription->get_currency();
 
 		if ( $discount ) {
-			$data[] = $this->format_invoice_item_data( -$discount, $subscription->get_currency(), __( 'Discount', 'woocommerce-payments' ) );
+			$data[] = $this->format_invoice_item_data( -$discount, $currency, __( 'Discount', 'woocommerce-payments' ) );
 		}
 
-		if ( $is_new ) {
-			// TODO: replace the line below once the subscription service has been copied over.
-			// $is_delayed = WCS_Stripe_Billing_Subscription_Service::has_delayed_payment( $subscription );.
-			$is_delayed = false;
-			$items      = $subscription->get_items();
```

Comment:
While testing various types of subscriptions, I'm noticing that Stripe invoice items are not being created for things like sign-up fees, shipping, etc. in Stripe. 

For example in WC, here is order confirmation for a subscription w/ a sign-up fee: https://cloudup.com/cXYPT5miY50

And the invoice for this in Stripe: https://cloudup.com/c9_jcj3kBCP

I'm actually not sure if these changes are specifically responsible, but I do recall needing to distinguish between new and recurring subscription payments to correctly get things like the sign-up fee, or to determine if we should in fact create invoice items in the first place in the case of delayed payment (free trials or synchronized subscriptions).

Were you able to work around this with these changes to `prepare_invoice_item_data`? If not, this may be the source of the problem?

Reply:
> Do you think this might also be playing a role in the issue you're noticing?

This is likely the culprit in the case of new subscriptions. So there are two methods for adding ad-hoc invoice items to a new subscription.

The first is to use the [add_invoice_items](https://stripe.com/docs/api/subscriptions/create#create_subscription-add_invoice_items) param when creating the subscription. This, unfortunately, requires us to have a Stripe product ID associated with whatever invoice item we are adding, which we decided would be impractical.

The second is what we are doing in the prototype, which is to first create the invoice items and attach them to the customer. These invoice items will then become associated with the next subscription created for that customer (the next step). I'm not sure how the `paid_out_of_band` approach might have impacted this at all though.

I did want to point out that it also looks like we aren't creating invoice items for renewals either, which should not be impacted by the removal of that line in `create_subscription`.

Reply:
> which is to first create the invoice items and attach them to the customer. These invoice items will then become associated with the next subscription created for that customer (the next step)

Aha, interesting! I didn't know this is how it works

I'll have to go and add that code back 😢 I'll do this in another PR I think.

Reply:
> I'll do this in another PR I think.

I finally found the issues! I needed to revert a lot of what I removed in this function (specifically `prepare_invoice_item_data()`) so I decided to just update this PR with the fixes and then I'll rebase the event handling on the latest of this branch.

The main issues were:
 - the `invoice/items` route we were using in `Payments API->create_invoice_item()` didn't exist (needed to be `invoices/items`)
 - the params we were sending to server to create invoice items wasn't correct (missing `invoiceitems` array key)
 - we were also using the `invoices/items` endpoint to create a single invoice item, but it should be used to create a batch
 - we incorrectly removed creating invoice items inside the `create_subscription()`

I've just pushed up the fixes to these issues.
While on this branch, when I create a subscription with shipping, the parent invoice in stripe will now have shipping.

Reply:
Nice work!

> the invoice/items route we were using in Payments API->create_invoice_item() didn't exist (needed to be invoices/items)

Ooof! I missed this in my review and should have noticed considering I did the server work for this. I'm sorry I didn't point it out. I hope this wasn't too much of a time sink for you 😰 

It looks like these changes did the trick and invoice items are being created again 🙌